src/mito2/src/config.rs

```diff
@@ -365,6 +370,13 @@ impl IndexConfig {
 
         Ok(())
     }
+
+    pub fn adjust_buffer_and_cache_size(&mut self, sys_memory: ReadableSize) {
+        let cache_size = cmp::min(sys_memory / MEM_CACHE_SIZE_FACTOR, ReadableSize::mb(128));
+        self.result_cache_size = cmp::min(self.result_cache_size, cache_size);
+        self.content_cache_size = cmp::min(self.content_cache_size, cache_size);
+        self.metadata_cache_size = cmp::min(self.metadata_cache_size, cache_size);
+    }
```

Comment:
We could use SST_META_CACHE_SIZE_FACTOR for the metadata_cache_size.

We already have so many caches. Maybe we should also use `1/32` as the default factor for index caches.